### PR TITLE
Recon Slice Naming

### LINKED ISSUE #2206: Improve Recon Slice Naming
### Desired Behaviour

Users would like to compare different reconstructed slices together. the workflow to achieve this currently is to reconstruct a slices using different filters, then open within the compare image stacks window. The saves slices have non-descriptive names such as Recon, Recon_2 etc. Each slice can't be renames easily within the left dialog of the main window meaning users can lose track of which recon is which.

To improve the descriptions of reconstructed slices, I propose we improve the default name to include the reconstruction method and filter so a user can quickly produce a number of different slices and still have a basic understanding of which reconstructed slice is which for comparison.

### Current Behaviour

Reconstructed slice when exported do not have a descriptive name making it hard to compare slices before reconstructing a volume.

### PR DESCRIPTION
**Pull request by JackEAllen**
### Issue

Closes #2206 

### Description

Modify recon output name to include if it is a slice, the algorithm name and filter name if a filter is available

### Testing 

- Reconstruct multiple individual slices with the same algorithm and filter
- Reconstruct slices with varying algorithms and filters and verify in the main window that names are easy to compare
- Reconstruct volume with the same algorithm and filter twice
- Reconstruct volumes with varying algorithms and filters and verify in the main window that names are easy to compare

### Acceptance Criteria 

*How should the reviewer test your changes*?
- Reconstructed slices follow the naming convention: Recon_slice_<Algorithm>_<Filter>
- Reconstructed volumes follow the naming convention: Recon_Vol_<Algorithm>_<Filter>
- If a filter is not present, not filter is added to filename

### Documentation

*How have you changed the documentation to reflect your changes? All changes should be noted in the appropriate file in docs/release_notes*

`docs/release_notes/next/feature-2206-recon_file_naming`

### CODE REVIEW

File: mantidimaging/gui/windows/recon/presenter.py [hunk 1]
```diff
@@ -289,7 +289,10 @@ def _on_stack_reconstruct_slice_done(self, task: TaskWorkerThread):
             slice_idx = self._get_slice_index(None)
             if images is not None:
                 assert self.model.images is not None
-                images.name = "Recon"
+                filter_name = self.view.recon_params().filter_name
+                algorithm_name = self.view.recon_params().algorithm
+                images.name = f"Recon_Slice{algorithm_name}"
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/recon/presenter.py:294]: Would be easier to read with an underscore, e.g `f"Recon_Slice_{algorithm_name}"`

File: mantidimaging/gui/windows/recon/presenter.py [hunk 2]
```diff
@@ -289,7 +289,10 @@ def _on_stack_reconstruct_slice_done(self, task: TaskWorkerThread):
...
+                images.name = f"{images.name}_{filter_name}" if filter_name else images.name
```
Comment (samtygier-stfc) [inline mantidimaging/gui/windows/recon/presenter.py:295]: The filter should only be added for gridrec and FBP, its not used by the iterative methods.

**Review by samtygier-stfc — CHANGES_REQUESTED** [review]

File: mantidimaging/core/reconstruct/cil_recon.py [hunk 3]
```diff
@@ -412,6 +412,6 @@ def full(images: ImageStack,
 
 def allowed_recon_kwargs() -> dict[str, list[str]]:
     return {
-        'CIL: PDHG-TV':
+        'CIL_PDHG-TV':
```
Comment (ashmeigh) [inline mantidimaging/core/reconstruct/cil_recon.py:415]: 'CIL_PDHG_TV' to be consistent

File: mantidimaging/gui/windows/recon/view.py [hunk 4]
```diff
@@ -90,7 +90,7 @@ def __init__(self, main_window: MainWindowView):
 
         self.algorithmName.insertItem(1, "FBP_CUDA")
         self.algorithmName.insertItem(2, "SIRT_CUDA")
-        self.algorithmName.insertItem(3, "CIL: PDHG-TV")
+        self.algorithmName.insertItem(3, "CIL_PDHG-TV")
```
Comment (ashmeigh) [inline mantidimaging/gui/windows/recon/view.py:93]: 'CIL_PDHG_TV' to be consistent

**Review by ashmeigh — CHANGES_REQUESTED** [review]
Completed the functional testing and reviewed the code. One small change requested.

File: mantidimaging/core/reconstruct/cil_recon.py (hunk 3 above)
Comment (JackEAllen) [reply mantidimaging/core/reconstruct/cil_recon.py:415]: I chose to use a hyphen between PDHG and TV as apposed to an underscore to show that TV is a configuration of the PDHG algorithm and directly related to the algorithm, and CIL is the library, not the algorithm. I am very happy to update if you feel that I should use an underscore for both 🙂

File: mantidimaging/gui/windows/recon/view.py (hunk 4 above)
Comment (JackEAllen) [reply mantidimaging/gui/windows/recon/view.py:93]: Same as above

Comment (JackEAllen) [reply mantidimaging/gui/windows/recon/view.py:93]: Same as above, happy to modify based on you're opinion to my comment above

File: mantidimaging/core/reconstruct/cil_recon.py (hunk 3 above)
Comment (ashmeigh) [reply mantidimaging/core/reconstruct/cil_recon.py:415]: That makes more sense.

**Review by ashmeigh — APPROVED** [review]
CIL_PDHG-TV: The hyphen between PDHG and TV indicates that TV is a configuration of the PDHG algorithm, with CIL representing the library. Therefore, using an underscore would be less appropriate.

File: mantidimaging/gui/windows/recon/presenter.py [hunk 5]
```diff
@@ -349,7 +352,9 @@ def _on_volume_recon_done(self, task) -> None:
         try:
             self._replace_inf_nan(task.result)  # pyqtgraph workaround
             assert self.model.images is not None
-            task.result.name = "Recon"
+            filter_name = self.view.recon_params().filter_name
+            task.result.name = f"Recon_Vol{self.view.recon_params().algorithm}"
+            task.result.name = f"{task.result.name}_{filter_name}" if filter_name else task.result.name
```
[the remark this answers is not in the document]
Comment (JackEAllen) [reply mantidimaging/gui/windows/recon/presenter.py]: I'll make a quick flash PR to put this in. when you approved, it merged it straight in XD

**Pull request merged**